### PR TITLE
Small fixes for clang + macosx

### PR DESCRIPTION
**Pull request by digantdesai**
Tested on vanilla macbook setup with the following

```
DISABLE_OPENMP=1 \
CC=$(which clang) \
TRITON_CPU_BACKEND=1 \
$(which python3) \
python/tutorials/01-vector-add.py
 ```

### CODE REVIEW

File: python/src/llvm.cc [hunk 1]
```diff
@@ -405,10 +414,18 @@ void init_triton_llvm(py::module &&m) {
...
+      // host
+      triple = llvm::sys::getProcessTriple();
+    }
+    mod->setTargetTriple(triple);
     std::string error;
     auto target =
         llvm::TargetRegistry::lookupTarget(mod->getTargetTriple(), error);
+    if (!target) {
+      throw std::runtime_error("target lookup error: " + error);
+    }
```
Comment (minjang) [inline python/src/llvm.cc:428]: The code is repetitive, maybe create a simple helper function?

File: python/src/llvm.cc [hunk 2]
```diff
@@ -435,8 +452,14 @@ void init_triton_llvm(py::module &&m) {
                 "failed to parse IR: " + error.getMessage() +
                 "lineno: " + std::to_string(error.getLineNo()));
           }
+
+          std::string triple = llvm::sys::getDefaultTargetTriple();
+          if (triple.empty()) {
+            // host
+            triple = llvm::sys::getProcessTriple();
+          }
```
Comment (minjang) [inline python/src/llvm.cc:460]: This also looks like can be refactored.

File: python/triton/runtime/build.py [hunk 3]
```diff
@@ -47,17 +47,24 @@ def _build(name, src, srcdir, library_dirs, include_dirs, libraries):
...
     cc_cmd += [f'-l{lib}' for lib in libraries]
     cc_cmd += [f"-L{dir}" for dir in library_dirs]
     cc_cmd += [f"-I{dir}" for dir in include_dirs if dir is not None]
     for dir in library_dirs:
         cc_cmd.extend(["-Wl,-rpath", dir])
     # CPU backend uses C++ (driver.cpp). Some old version compilers need a specific C++17 flag.
     if src.endswith(".cpp") or src.endswith(".cc"):
-        cc_cmd += ["-std=c++17", "-fopenmp"]
+        cc_cmd += ["-std=c++17"]
+        if not os.environ.get("DISABLE_OPENMP", None):
```
Comment (minjang) [inline python/triton/runtime/build.py:66]: Is `DISABLE_OPENMP` only for triton? Then I'd suggest `TRITON_DISABLE_OPENMP` or `TRITON_DISABLE_OMP`.

**Review by minjang — APPROVED** [review]
Looks good to me! Just a minor refactoring suggestion.

File: python/src/llvm.cc (hunk 1 above)
Comment (digantdesai) [reply python/src/llvm.cc]: Moved the `triple` population in a subroutine.

File: python/triton/runtime/build.py (hunk 3 above)
Comment (digantdesai) [reply python/triton/runtime/build.py]: renamed.

**Pull request merged**